gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementTaskStreamImpl.java

```diff
@@ -220,17 +230,17 @@ private DagTask createDagTask(DagActionStore.DagAction dagAction, LeaseAttemptSt
 
     switch (dagActionType) {
       case ENFORCE_FLOW_FINISH_DEADLINE:
-        return new EnforceFlowFinishDeadlineDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore);
+        return new EnforceFlowFinishDeadlineDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore, dagProcEngineMetrics);
       case ENFORCE_JOB_START_DEADLINE:
-        return new EnforceJobStartDeadlineDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore);
+        return new EnforceJobStartDeadlineDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore, dagProcEngineMetrics);
       case KILL:
-        return new KillDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore);
+        return new KillDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore, dagProcEngineMetrics);
       case LAUNCH:
-        return new LaunchDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore);
+        return new LaunchDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore, dagProcEngineMetrics);
       case REEVALUATE:
-        return new ReevaluateDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore);
+        return new ReevaluateDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore, dagProcEngineMetrics);
       case RESUME:
-        return new ResumeDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore);
+        return new ResumeDagTask(dagAction, leaseObtainedStatus, dagManagementStateStore, dagProcEngineMetrics);
```

Comment:
this does not look good but sadly i do not have any easy alternate suggestion for dag tasks to have dagProcEngineMetrics